v0.43.0 fails to compile with default features: FormatItem::StringLiteral does not exist

`lopdf 0.43.0` does not compile with its default feature set. The `time` feature is enabled by default, and `src/datetime.rs`'s `time_impl` module references a `time`-crate API that doesn't exist.

## Reproduction

```sh
cargo new repro && cd repro
cargo add lopdf@0.43
cargo build
```

## Error

```
error[E0599]: no variant, associated function, or constant named `StringLiteral`
found for enum `BorrowedFormatItem<'a>`
  --> lopdf-0.43.0/src/datetime.rs:103
      date.format(&FormatItem::StringLiteral("%Y%m%d%H%M%SZ")).unwrap()
                               ^^^^^^^^^^^^^
```

## Cause

In `impl From<Time> for Object` (`src/datetime.rs`, inside `#[cfg(feature = "time")] mod time_impl`):

```rust
date.format(&FormatItem::StringLiteral("%Y%m%d%H%M%SZ")).unwrap()
```

Two distinct problems:

1. `FormatItem` / `BorrowedFormatItem` has no `StringLiteral` variant (its variants are `Literal`, `Component`, `Compound`, `Optional`, `First`).
2. `"%Y%m%d%H%M%SZ"` is strftime syntax; the `time` crate uses `[year][month]…` component descriptions — see the adjacent `impl From<OffsetDateTime>`, which is correct. A `time::Time` also has no date components, so `%Y%m%d` can't apply to it regardless.

The neighboring `impl From<OffsetDateTime>` directly below does it correctly with `time::format_description::parse_borrowed::<3>("…[year][month]…")`, which suggests the `From<Time>` impl is leftover/mis-ported code that never compiled under this feature.

## Suggested fix

Rework `impl From<Time> for Object` to use `time::format_description::parse_borrowed` like the neighboring `From<OffsetDateTime>` impl (formatting only the available time components), or drop the impl entirely. Adding a CI job that builds with default features (or `--all-features`) would catch this class of regression — cf. #226, which was the same category of bug ("fails to build with default_features = false").

## Note on `time` source ("works on my machine")

This resolves the standard crates.io `time` (`^0.3`, `formatting` + `parsing` features) — lopdf's own published manifest declares no `git`/`path`/patched source for it. If it builds in your workspace, you may have a local `[patch.crates-io]` or fork of `time`; the released crate uses stock `time`, where `FormatItem`/`BorrowedFormatItem` has no `StringLiteral` variant (confirmed against `time` 0.3.36 and 0.3.41). A CI job building with default features on a clean lockfile would catch this.

## Workaround for downstream users

Disable default features and skip `time`:

```toml
lopdf = { version = "0.43", default-features = false, features = ["chrono", "jiff", "rayon"] }
```

--- a/src/datetime.rs
+++ b/src/datetime.rs
@@ -92,18 +92,26 @@ mod jiff_impl {
 #[cfg(feature = "time")]
 mod time_impl {
     use crate::Object;
-    use time::{OffsetDateTime, Time, format_description::FormatItem};
+    use time::{OffsetDateTime, PrimitiveDateTime};
 
-    impl From<Time> for Object {
-        fn from(date: Time) -> Self {
-            // can only fail if the TIME_FMT_ENCODE_STR would be invalid
-            Object::string_literal(
-                format!(
-                    "D:{}",
-                    date.format(&FormatItem::StringLiteral("%Y%m%d%H%M%SZ")).unwrap()
+    /// The naive datetime is taken to be UTC and rendered with the `Z` suffix
+    /// (PDF 32000-1 §7.9.4) — the `time`-crate counterpart of the
+    /// `chrono::DateTime<Utc>` and `jiff::Timestamp` impls above.
+    ///
+    /// (This replaces a `From<time::Time>` impl that never compiled — see
+    /// issue #518: it referenced a nonexistent `FormatItem::StringLiteral`
+    /// with a strftime pattern, and a bare time-of-day cannot form a valid
+    /// PDF date anyway, which must start at the year.)
+    impl From<PrimitiveDateTime> for Object {
+        fn from(date: PrimitiveDateTime) -> Self {
+            Object::string_literal({
+                // D:%Y%m%d%H%M%SZ
+                let format = time::format_description::parse_borrowed::<3>(
+                    "D:[year][month][day][hour][minute][second]Z",
                 )
-                .into_bytes(),
-            )
+                .unwrap();
+                date.format(&format).unwrap()
+            })
         }
     }
 
@@ -253,6 +261,22 @@ fn parse_datetime_time_missing_jiff() {
     let text = Object::string_literal("D:20040229");
     let dt: Option<Zoned> = text.as_datetime().and_then(|dt| dt.try_into().ok());
     assert!(dt.is_some());
+}
+
+#[cfg(feature = "time")]
+#[test]
+fn format_primitive_datetime_as_utc() {
+    use time::{Date, Month, PrimitiveDateTime, Time};
+
+    // The example date from the PDF reference (1.7, §3.8.3), as a naive
+    // datetime: it must serialize with the year first and the `Z` suffix.
+    let date = Date::from_calendar_date(1998, Month::December, 23).unwrap();
+    let time = Time::from_hms(19, 52, 0).unwrap();
+    let text: Object = PrimitiveDateTime::new(date, time).into();
+    match &text {
+        Object::String(bytes, _) => assert_eq!(bytes.as_slice(), b"D:19981223195200Z"),
+        other => panic!("expected a string literal, got {other:?}"),
+    }
 }
 
 #[cfg(feature = "time")]

